packages/calypso-e2e/src/element-helper.ts

```diff
@@ -75,3 +75,30 @@ export async function clickNavTab( page: Page, name: string ): Promise< void > {
 	const elementHandle = await page.waitForSelector( selectors.navTab( name ) );
 	await Promise.all( [ page.waitForNavigation(), elementHandle.click() ] );
 }
+
+/**
+ * Retry any action up to three times or action passes without throwing an exception,
+ * whichever comes first.
+ *
+ * This is useful for situations where the backend must process the results of a
+ * previous action then inform the front end of the result of the process.
+ * An example of this is the user invitation system where the backend must receive,
+ * verify and send the invitation issued by a user. If the resulting checks were
+ * successful, the resulting invitation is shown as 'pending'.
+ *
+ * @param {Page} page Page object.
+ */
+export async function retryAction(
+	page: Page,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	func: ( page: Page, ...args: any[] ) => Promise< void >
+): Promise< void > {
+	for ( let retries = 3; retries > 0; retries -= 1 ) {
+		try {
+			await func( page );
+		} catch {
+			await page.reload();
+		}
```

Comment:
Do you think we should have the action throw the last time? If this hasn't worked, it's busted, and we probably want the test to fail I think. We don't have to, but it seems to fit the Playwright pattern a bit better.

Reply:
Fixed in 78d067d242bf8c2e0f9d2e842a9e2f4b84497b15.